src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -121,23 +168,41 @@ const useFetchChannelHook = () => {
   const saveAllChannelData = async (channels, channelCategory: ChannelCategory) => {
     const filteredChannels = filterChannels(channels);
 
-    filteredChannels?.forEach(async (channel) => {
-      await saveChannelData(channel, channelCategory);
+    const channelPromises = filteredChannels?.map((channel) => {
+      return new Promise((resolve, reject) => {
+        try {
+          saveChannelData(channel, channelCategory).then(() => resolve(true));
+        } catch (error) {
+          reject(error);
+        }
+      });
+    });
+
+    queue.addJob({
+      label: 'timestamp update',
+      task: async () => {
+        const timestamp = new Date().toISOString();
+        StorageUtils.channelSignedTimeStamps.set(timestamp);
+      }
     });
+
+    await Promise.all(channelPromises);
   };
 
   const getAllSignedChannels = async () => {
     if (!localDb) return;
     let signedChannel;
 
     try {
-      signedChannel = await SignedMessageRepo.getAllSignedChannels();
+      const timeStamp = StorageUtils.channelSignedTimeStamps.get();
+      signedChannel = await SignedMessageRepo.getAllSignedChannels(timeStamp as string);
     } catch (e) {
       console.log('error on getting signedChannel:', e);
     }
 
     try {
-      await saveAllChannelData(signedChannel ?? [], 'SIGNED');
+      if (Array.isArray(signedChannel) && signedChannel.length === 0) return;
+      await saveAllChannelData(signedChannel, 'SIGNED');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes here introduce the use of a timestamp for fetching signed channels. This can potentially improve performance by reducing the amount of data fetched from the server. However, it's important to ensure that this change doesn't introduce issues with data consistency. Consider adding tests to verify that all expected channels are correctly fetched and saved.



<!-- This is an auto-generated comment by OSS CodeRabbit -->